UIControlPublisher.playground/Sources/UIControlPublisher.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  UIControlPublisher.swift
+//  UIControlPublisher
+//
+//  Created by Alfonso on 05/10/2019.
+//  Copyright © 2019 Alfonso. All rights reserved.
+//
+
+import Combine
+import UIKit
+
+public struct UIControlPublisher<Control: UIControl>: Publisher {
+    public typealias Output = Control
+    public typealias Failure = Never
+    
+    private let control: Control
+    private let event: UIControl.Event
+
+    init(control: Control, event: UIControl.Event) {
+        self.control = control
+        self.event = event
+    }
+    
+    // The Publisher only creates a new subscription and pass it to the subscriber
+    public func receive<S>(subscriber: S) where S : Subscriber, Failure == S.Failure, Output == S.Input {
+        let subscription = UIControlSubscription(subscriber: subscriber, control: control, event: event)
+        subscriber.receive(subscription: subscription)
+    }
+}
+
+// MARK: Convenience
+
+public protocol UIControlPublisherBuilder: UIControl { }
+
+public extension UIControlPublisherBuilder {
```

Comment:
why not making it directly to `extension UIControl`?
Do you want to attach it to instances different of UIControl?

Reply:
Just a trick that allows the type of the generic publisher to be more specific:

`func publisher(for event: UIControl.Event) -> UIControlPublisher<Self>`

With a UIControl extension you can only return a less specific publisher: 
`UIControlPublisher<UIControl>`

Example:
```
let button = UIButton(frame:  .zero)
let publisher = button.publisher(for: .touchUpInside) // type: UIControlPublisher<UIButton>
```